kyaml/kio/filters/fmtr_test.go

```diff
@@ -904,3 +904,93 @@ func TestFormatFileOrDirectory_trimWhiteSpace(t *testing.T) {
 
 	assert.Equal(t, string(testyaml.FormattedYaml1), string(b))
 }
+
+func TestFormatFileOrDirectory_FmtAnnotation(t *testing.T) {
```

Comment:
Why not have yaml in yaml out tests so that it's more readable.

Reply:
It would require adding several new examples to cover all the cases I wanted to test. So just updating the single field that would change seems better in this case.

Reply:
Agree. But not sure if it's good to have code within tests to edit yaml, as that code(addFmtAnnotation) itself is not tested. And it restricts other developers to use single yaml here. 

The other case I could think of is one file having 2 resources one with annotation and other without it and see if the output is as expected, which might be tough to add it in future/now with this approach.

Reply:
i like the overall approach, but agree with Phani that all the code just to set an annotation makes this test harder to read for someone new, and the pattern will be copied.

Can we factor out a test utility that holds what's in common 
between `addFmtAnnotation` and, say,
https://github.com/kubernetes-sigs/kustomize/blob/master/api/testutils/kusttest/harnessenhanced.go#L148

so that this and other  tests using it are easier to read?

I think then its okay to just call it as an untested utility function;
it's implicitly tested by being used in so many tests.

Reply:
I updated the PR by splitting the code shared with `harnessechanced` into a separate utility function.
I think this test is ok, but the main way we could make the code simpler is to just add the input and output yaml into the test itself. It means quite a bit more code, but less complexity. Thoughts?

Reply:
i'll take more LOC over complexity any day.

I'd rather see a bunch of inline YAML here, and no disk use.

Not picking on this test in particular, just generally.

Reply:
Updated the test to inline the yaml. I like this better than the previous version.